regulation-worker/internal/delete/api/api.go

```diff
@@ -106,22 +105,29 @@ func (api *APIManager) Delete(ctx context.Context, job model.Job, destination mo
 		return model.JobStatusFailed
 	}
 	jobStatus := getJobStatus(resp.StatusCode, jobResp)
-	pkgLogger.Debugf("[%v] JobStatus for %v: %v", destination.Name, destination.DestinationID, jobStatus)
+	pkgLogger.Debugf("[%v] JobStatus for %v: %v", destination.Name, job.DestinationID, jobStatus)
 
-	if isOAuthEnabled && isTokenExpired(jobResp) {
+	if isOAuthEnabled && isTokenExpired(jobResp) && attempts < 1 {
 		err = api.refreshOAuthToken(destination.Name, job.WorkspaceID, oAuthDetail)
 		if err != nil {
 			pkgLogger.Error(err)
 			return model.JobStatusFailed
 		}
 		// retry the request
-		pkgLogger.Debug("Retrying deleteRequest job for the whole batch")
-		return api.Delete(ctx, job, destination)
+		pkgLogger.Debugf("Retrying deleteRequest job for the whole batch: %v", job.DestinationID)
+		attempts++
```

Comment:
It doesn't make sense to increment the incoming variable. You can send `attempts + 1` during invocation.

Reply:
Done. Can you please verify ?